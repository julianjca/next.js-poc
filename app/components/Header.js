import Link from 'next/link'

const linkStyle = {
  marginRight: 15
}

const Header = (props) => (
  <div>
    <h2>{props.title}</h2>
    <hr />
    <Link href="/">
      <a style={linkStyle}>Home</a>
    </Link>
    <Link href="/about">
      <a style={linkStyle}>About</a>
    </Link>
  </div>
)

export default Header