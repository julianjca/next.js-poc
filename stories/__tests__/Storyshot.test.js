import initStoryshots, { shallowSnapshot } from '@storybook/addon-storyshots';
import shallowRenderer from 'react-test-renderer/shallow'
import Header from '../../app/components/Header'

initStoryshots({
    // test: shallowSnapshot
    // test({ story, context }) {
    //     const renderer = new shallowRenderer
    //     const result = renderer.render(story.render(context))
    //     expect(result).toMatchSnapshot()
    // }
    test: ({ story, context }) => {
        // console.log(story.render(context).props.initialContent)
        const shallow = new shallowRenderer
        // const code = <Header title="OK" />
        // console.log(code)
        shallow.render(story.render(context).props.initialContent)
        expect(shallow.getRenderOutput()).toMatchSnapshot();
    }
});