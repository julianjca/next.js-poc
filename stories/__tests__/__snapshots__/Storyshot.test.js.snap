// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout Header 1`] = `
<div>
  <h2>
    ทดสอบ
  </h2>
  <hr />
  <Link
    href="/"
  >
    <a
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
  </Link>
  <Link
    href="/about"
  >
    <a
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </Link>
</div>
`;

exports[`Storyshots Layout Index 1`] = `
<Layout>
  <h1>
    My Blog
  </h1>
  <ul>
    <PostLink
      post={
        Object {
          "id": 1,
          "name": "test",
        }
      }
    />
    <PostLink
      post={
        Object {
          "id": 2,
          "name": "test2",
        }
      }
    />
  </ul>
</Layout>
`;
